lib/core/src/Network/Wai/Middleware/ServantError.hs

```diff
@@ -86,3 +87,7 @@ responseBody = \case
         Nothing
     ResponseStream{} ->
         Nothing
+
+servantErrorMsg :: String -> String -> String
+servantErrorMsg expected received =
+    "expected " <> expected <> ", encountered " <> received
```

Comment:
**[remark]**

Hey! I've got several things for which I have mixed feelings here ^.^ 

- Such function doesn't feel in the right place. The module provides a Wai middleware for wrapping raw servant error into JSON objects. But it does little care about error messages so to speak. A better home would probably be: https://github.com/input-output-hk/cardano-wallet/blob/master/lib/core-integration/src/Test/Integration/Framework/TestData.hs#L49

- This error "expected X, encountered Y" is actually not coming from `servant` but it's coming from `aeson`. See [typeMismatch](https://hackage.haskell.org/package/aeson-1.4.5.0/docs/Data-Aeson-Types.html#v:typeMismatch)

- The signature is very much stringly-typed. If introducing anything like this, I'd suggest to reuse `typeMismatch` from Aeson, and perhaps, provide an extra function in addition to convert a `Parser a -> String` though that feels a bit weird. 

- A better fix would actually be to either:
    - find a way to not assert error messages (though with Aeson is very much stringly typed as well :s)
    - Or, actually define a test helper function to perform these tests! So, instead of 

    ```hs
        it "ApiT AddressPoolGap (not a integer)" $ do
            let msg = "Error in $: expected Integer, encountered floating number\
                    \ 2.5"
            let msg = "Error in $: " <> servantErrorMsg "Integer" "floating number 2.5"
            Aeson.parseEither parseJSON [aesonQQ|
                2.5
            |] `shouldBe` (Left @String @(ApiT AddressPoolGap) msg)
    ```

    one could write:

    ```hs
    testAesonParseError (Proxy (ApiT AddressPoolGap)) (toJSON 2.5) 
    ```

    The _right_ error message could probably be inferred from an `Arbitrary` instance on the given type, and a `ToJSON`. If that turns out to be too complicated, then we could imagine give the expected string type as an argument.

Reply:
@KtorZ thanks! yeah, of course it comes from aeson :upside_down_face: I will try to go into `testAesonParseError` direction